### PR TITLE
enhance: [2.4] Improve GetVectorById of Sparse Float Vector

### PR DESCRIPTION
**Pull request by zhengbuqian**
issue: https://github.com/milvus-io/milvus/issues/29419
pr: #33209 

codecov will fail due to newly added ut in test_sealed.cpp skipped due to https://github.com/milvus-io/milvus/issues/33210

### CODE REVIEW

**Review by liliu-z — COMMENTED** [review]
/lgtm

File: internal/core/src/mmap/Column.h [hunk 1]
```diff
@@ -153,11 +158,18 @@ class ColumnBase {
...
+    // Data() points at an addr that contains the elements
     virtual const char*
     Data() const {
         return data_;
     }
 
+    // MmappedData() returns the mmaped address
+    const char*
+    MmappedData() const {
+        return data_;
```
Comment (liliu-z) [inline internal/core/src/mmap/Column.h:170]: Any difference?

**Review by liliu-z — APPROVED** [review]
/lgtm

**Review by congqixia — APPROVED** [review]
/lgtm

File: internal/core/src/mmap/Column.h (hunk 1 above)
Comment (zhengbuqian) [reply internal/core/src/mmap/Column.h:170]: pasting my reply in #33209 here:

`Data()` points at an array of elements, while `MmappedData()` points at the entire mmap-ed memory. They are the same for dense vectors but not the same for sparse vectors. SparseFloatColumn overrides `Data()` to return something different.

**Pull request merged**